app/models/agents/pushbullet_agent.rb

```diff
@@ -114,14 +109,17 @@ def create_device
       end
     end
 
-
     def basic_auth
       {basic_auth: {username: interpolated[:api_key].presence || credential('pushbullet_api_key'), password: ''}}
     end
 
     def query_options(event)
       mo = interpolated(event)
-      basic_auth.merge(body: {device_iden: mo[:device_id], type: mo[:type]}.merge(payload(mo)))
+      if mo[:device_id] == "__ALL__"
+        basic_auth.merge(body: {device_iden: "", type: mo[:type]}.merge(payload(mo)))
```

Comment:
The outer if adds a bit of code duplication, you could inline the condition
```
basic_auth.merge(body: {device_iden: (mo[:device_id] == "__ALL__" ? mo[:device_id] : '') , type: mo[:type]}.merge(payload(mo)))
```
or use a `device_iden` variable to avoid long, complex line
```
device_iden = mo[:device_id] == "__ALL__" ? mo[:device_id] : ''
basic_auth.merge(body: {device_iden: device_iden, type: mo[:type]}.merge(payload(mo)))
```